### PR TITLE
Updated to nullsafety with the latest and greatest (injectable, icapp…

### PR DESCRIPTION
**Pull request by ikbendewilliam**
Updated to nullsafety with the latest and greatest (injectable, icapos_architecture)

### CODE REVIEW

File: bin/icapps_generator.dart [hunk 1]
```diff
@@ -3,31 +3,30 @@ import 'dart:io';
...
 import 'src/file_creator_helper.dart';
-import 'src/flutter_helper.dart';
 import 'src/params.dart';
 
 var screenName = '';
-Params params;
+late final Params params;
 const NO_NAV_ARG = '--no-nav';
-const NO_KIWI_ARG = '--no-kiwi';
+const NO_INJECTABLE_ARG = '--no-di';
```
Comment (vanlooverenkoen) [inline bin/icapps_generator.dart:11]: your idea of moving to --no-di is indeed more generic. But I think we should use an other constant name. NO_DI_ARG for example

File: bin/icapps_generator.dart [hunk 2]
```diff
@@ -36,26 +35,22 @@ Future<void> main(List<String> args) async {
...
-  print('Generate Kiwi Dependency Tree: $generateKiwi');
+  print('Generate injectable: $generateInjectable');
   print('\n');
   print('\n');
   print('Generating a new screen called `$screenName`');
   createFolders();
-  createFiles();
-  await FileCreatorHelper.updateInjector(params.projectName, screenName);
+  createFiles(generateInjectable: generateInjectable);
+  // await FileCreatorHelper.updateInjector(params.projectName, screenName);
```
Comment (vanlooverenkoen) [inline bin/icapps_generator.dart:49]: is this comment still needed

File: bin/src/file_creator_helper.dart [hunk 3]
```diff
@@ -8,122 +8,80 @@ import 'case_util.dart';
...
+    } else {
+      sb.writeln();
+    }
+    sb
+      ..writeln('class ${CaseUtil.getCamelcase(screenName)}ViewModel with ChangeNotifierEx {')
+      ..writeln('  late final ${CaseUtil.getCamelcase(screenName)}Navigator _navigator;')
       ..writeln()
-      ..writeln(
-          '  Future<void> init(${CaseUtil.getCamelcase(screenName)}Navigator navigator) async {')
+      ..writeln('  Future<void> init(${CaseUtil.getCamelcase(screenName)}Navigator navigator) async {')
```
Comment (vanlooverenkoen) [inline bin/src/file_creator_helper.dart:22]: format code

File: example/.flutter-plugins [hunk 4]
```diff
@@ -0,0 +1,16 @@
+# This is a generated file; do not edit or check into version control.
```
Comment (vanlooverenkoen) [inline example/.flutter-plugins:1]: do not check into version controll

File: example/.flutter-plugins-dependencies [hunk 5]
```diff
@@ -0,0 +1 @@
..._preferences_web-2.0.0/","dependencies":[]}]},"dependencyGraph":[{"name":"connectivity_plus","dependencies":["connectivity_plus_linux","connectivity_plus_macos","connectivity_plus_web","connectivity_plus_windows"]},{"name":"connectivity_plus_linux","dependencies":[]},{"name":"connectivity_plus_macos","dependencies":[]},{"name":"connectivity_plus_web","dependencies":[]},{"name":"connectivity_plus_windows","dependencies":[]},{"name":"device_info_plus","dependencies":["device_info_plus_macos","device_info_plus_web"]},{"name":"device_info_plus_macos","dependencies":[]},{"name":"device_info_plus_web","dependencies":[]},{"name":"path_provider_linux","dependencies":[]},{"name":"path_provider_windows","dependencies":[]},{"name":"shared_preferences","dependencies":["shared_preferences_linux","shared_preferences_macos","shared_preferences_web","shared_preferences_windows"]},{"name":"shared_preferences_linux","dependencies":["path_provider_linux"]},{"name":"shared_preferences_macos","dependencies":[]},{"name":"shared_preferences_web","dependencies":[]},{"name":"shared_preferences_windows","dependencies":["path_provider_windows"]}],"date_created":"2021-05-12 18:39:00.833472","version":"2.0.5"}
```
Comment (vanlooverenkoen) [inline example/.flutter-plugins-dependencies:1]: do not check into version controll

File: example/lib/navigator/main_navigator.dart [hunk 6]
```diff
@@ -1,31 +1,25 @@
...
-    final MainNavigatorWidgetState navigator = rootNavigator
-        ? context.rootAncestorStateOfType(
-            const TypeMatcher<MainNavigatorWidgetState>(),
-          )
-        : context.ancestorStateOfType(
-            const TypeMatcher<MainNavigatorWidgetState>(),
-          );
+  static MainNavigatorWidgetState of(BuildContext context, {bool rootNavigator = false}) {
+    final navigator = rootNavigator ? context.findRootAncestorStateOfType<MainNavigatorWidgetState>() : context.findAncestorStateOfType<MainNavigatorWidgetState>();
     assert(() {
```
Comment (vanlooverenkoen) [inline example/lib/navigator/main_navigator.dart:14]: assert can be replaced with simple if

**Review by vanlooverenkoen — CHANGES_REQUESTED** [review]

**Comment by ikbendewilliam** [comment]
Taken over and new PR created by colleague

**Pull request closed without merging**